control-plane/roles/gardener/templates/shoot-dns-service/controller-deployment.yaml

```
@@ -10,10 +10,11 @@ providerConfig:
     image:
       repository: "{{ gardener_shoot_dns_service_image_name }}"
       tag: "{{ gardener_shoot_dns_service_image_tag }}"
+{% if gardener_shoot_dns_service_image_vector_overwrite %}
+    imageVectorOverwrite:
+      images: "{{ gardener_shoot_dns_service_image_vector_overwrite | to_json }}"
```

Why not like `network-cilium`?
```suggestion
      images:
        "{{ gardener_shoot_dns_service_image_vector_overwrite | to_nice_yaml(indent=2) | indent(width=8, first=false }}"
```

---

I cannot test it. If you are sure it works, we can do it.

---

Added a test now. You missed a parenthesis but apart from this it worked fine. :)